### PR TITLE
feat(home-manager): init wlogout module

### PR DESCRIPTION
**Pull request by Anomalocaridid**
Adds support for wlogout.

Extra options I added:
- `iconStyle`, which lets the user specify whether to use the wlogout or wleave-style icons that come with the theme.
- `extraStyle`, which lets the user add extra CSS to ~/.config/wlogout.css

In order for the svg icons to display properly, [`pkgs.librsvg`](https://search.nixos.org/packages?channel=unstable&show=librsvg&from=0&size=50&sort=relevance&type=packages&query=librsvg) needs to be in [`programs.gdk-pixbuf.modulePackages`](https://search.nixos.org/options?channel=unstable&show=programs.gdk-pixbuf.modulePackages&from=0&size=50&sort=relevance&type=packages&query=gdk-pixbuf), but I am not sure if I should add an assertion or warning or something or not.

### CODE REVIEW

**Review by isabelroses — COMMENTED** [review]
If it needs additional packages it maybe worth waiting for https://github.com/catppuccin/nix/pull/384.

I would also prefer not to merge any PRs adding new modules until https://github.com/catppuccin/nix/pull/371 is merged too.

**Comment by Anomalocaridid** [comment]
@isabelroses Makes sense. In that case, I will mark this as a draft for now.

**Comment by Anomalocaridid** [comment]
I have updated this PR to reflect the recent treewide changes.

File: modules/home-manager/wlogout.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
+{ catppuccinLib }:
+{ config, lib, ... }:
+
+let
+  inherit (config.catppuccin) sources;
+
+  cfg = config.catppuccin.wlogout;
+  enable = cfg.enable && config.programs.wlogout.enable;
```
Comment (isabelroses) [inline modules/home-manager/wlogout.nix:8]: We have started to remove this from modules that use `.enable` like so 
https://github.com/nix-community/home-manager/blob/10e99c43cdf4a0713b4e81d90691d22c6a58bdf2/modules/programs/wlogout.nix#L73
```suggestion
```

File: modules/home-manager/wlogout.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          }
+
+          #lock {
+            background-image: url("''${config.gtk.iconTheme.package}/share/icons/''${config.gtk.iconTheme.name}/apps/scalable/system-lock-screen.svg");
+          }
+        '';
+      };
+    };
+
+  config = lib.mkIf enable {
```
Comment (isabelroses) [inline modules/home-manager/wlogout.nix:43]:
```suggestion
  config = lib.mkIf cfg.enable {
```

**Review by isabelroses — CHANGES_REQUESTED** [review]

**Review by isabelroses — APPROVED** [review]

**Pull request merged**